crates/dojo-core/src/world/world_contract.cairo

```diff
@@ -284,10 +312,31 @@ pub mod world {
     }
 
     #[generate_trait]
-    impl ResourceIsNoneImpl of ResourceIsNoneTrait {
-        fn is_unregistered(self: @Resource) -> bool {
+    impl ResourceTypeImpl of ResourceTypeTrait {
+        fn is_unregistered(self: @ResourceType) -> bool {
             match self {
-                Resource::Unregistered => true,
+                ResourceType::Unregistered => true,
+                _ => false
+            }
+        }
+
+        fn is_model(self: @ResourceType) -> bool {
+            match self {
+                ResourceType::Model => true,
+                _ => false
+            }
```

Comment:
`if let` might be more idiomatic to test a single case.
```rust
if let ResourceType::X(_) = self { true } else { false }
```

Reply:
Nice !